bin/plot_truncated_pexecs

```diff
@@ -202,13 +202,24 @@ def generate_trials(json_file):
     print 'Start parallel execution.'
     # start_ends describes which work is done on which core, and must match the
     # physical machine. Intervals are [inclusive, exclusive).
-    start_ends = [(2, 11), (10, 21), (20, 31)]  # 4-core machine.
-    # start_ends = [(2, 3), (3, 4), (4, 5), (5, 6),  # 32-core machine.
-    #               (6, 7), (7, 8), (9, 10), (10, 11),
-    #               (11, 12), (12, 13), (13, 14), (14, 15), (15, 16),
-    #               (16, 17), (17, 18), (18, 19), (19, 20), (20, 21),
-    #               (21, 22), (22, 23), (23, 24), (24, 25), (25, 26),
-    #               (26, 27), (27, 28), (28, 29), (29, 30), (30, 31)]
+    # The second number in the last tuple in the list must by 1 + MAX_PEXECS.
+    assert cores in (4, 8, 16, 32), 'Invalid number of cores (%d)' % cores
```

Comment:
I think we need a more robust error message: `assert`s can be optimised away. So something like:

```python
if cores not in [4, 8, 16, 32]:
    raise ...
```

or `sys.stderr.write(...)` or similar.